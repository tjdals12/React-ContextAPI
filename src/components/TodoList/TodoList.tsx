import React from 'react';
import classNames from 'classnames/bind';
import TodoItem from 'components/TodoItem';
import styles from './TodoList.scss';

const cx = classNames.bind(styles);

type Todo = {
    id: number;
    text: string;
    done: boolean;
};

type TodoListProps = {
    todos: Todo[];
};

function TodoList({ todos }: TodoListProps): JSX.Element {
    return (
        <ul className={cx('TodoList')}>
            {todos.map(todo => (
                <TodoItem key={todo.id} todo={todo} />
            ))}
        </ul>
    );
}

export default TodoList;
