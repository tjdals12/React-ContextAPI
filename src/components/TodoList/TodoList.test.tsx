import React from 'react';
import { render } from '@testing-library/react';
import TodoList from 'components/TodoList';

describe('<TodoList />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoList todos={[]} />);
        expect(utils.container).toMatchSnapshot();
    });
});
