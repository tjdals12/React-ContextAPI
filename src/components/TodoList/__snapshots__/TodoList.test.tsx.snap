// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<div>
  <ul
    class="TodoList"
  />
</div>
`;
